front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx

```diff
@@ -144,7 +144,11 @@ export default function AssistantBuilderDataSourceModal({
         }
       }}
       onSave={() => onSaveLocal({ isSelectAll })}
-      hasChanged={selectedDataSource !== null}
+      hasChanged={
```

Comment:
This will Save the change even if you click on cancel. We need to change the onSave logic 🙈